modules/classify_novel_taxa.nf

```diff
@@ -0,0 +1,448 @@
+	"""
+	spades.py --rna -s ${fastq} -t ${task.cpus} -o rnaspades
+	"""
+}
+
+
+process assemble_megahit_paired {
+	
+	label "process_high"
+
+	conda "bioconda::megahit"
+	container "biowilko/scylla@${params.wf.container_sha}"
+
+    input:
+        val unique_id
+        path fastq_1
+        path fastq_2
+
+    output:
+        path "megahit/final.contigs.fa"
+
+	script:
+	"""
+	megahit -1 ${fastq_1} -2 ${fastq_2} \
+        	-m 0.5 --min-contig-len 150 --presets meta-sensitive -t 8 -o megahit
+	"""
+}
+
+process assemble_rnaspades_paired {
+
+	label "process_high"
+
+	//----------ENV-----------
+	container "biowilko/scylla@${params.wf.container_sha}"
+
+    input:
+        val unique_id
+        path fastq_1
+        path fastq_2
+
+    output:
+        path "rnaspades/transcripts.fasta"
+
+	script:
+	"""
+	spades.py --rna -1 ${fastq_1} -2 ${fastq_2} -t ${task.cpus} -o rnaspades
+	"""
+}
+
+
+
+// 1a. Count assembly statistics
+
+process gen_assembly_stats {
+
+	label "process_single"
+
+    conda "bioconda::bbmap"
+	container "biowilko/scylla@${params.wf.container_sha}"
+
+    publishDir "${params.out_dir}/${unique_id}/discovery", mode: 'copy', saveAs: { filename -> "${params.assembler}_${filename}" }
+
+    input:
+	    val unique_id
+	    path contigs
+
+    output:
+        path("assembly_stats.txt")
+
+    script:
+    """
+    stats.sh in=${contigs} out="assembly_stats.txt"
+    """
+}
+
+// 2. Run identification
+
+process run_virbot {
+
+	label "process_medium"
+
+        // UPLOAD ENV - DELETE WHEN SET!!!
+	conda "/localdisk/home/s2420489/conda/virbot.yml"
+	container "biowilko/scylla@${params.wf.container_sha}"
+
+    publishDir "${params.out_dir}/${unique_id}/discovery", mode: 'copy', saveAs: { it == "virbot/output.vb.fasta" ? "viral_contigs.fa" : "tax_assignments.tsv" }
+
+    input:
+        val unique_id
+	    path contigs
+
+    output:
+        path "virbot/output.vb.fasta"
+	    path "pos_contig_score.tsv"
+
+    script:
+    """
+    virbot --input "${contigs}" --output virbot
+	awk -F',' 'BEGIN {print "contig_id\ttaxonomy"} NR>1{ gsub(/; /, ";", \$4) ; print \$1"\t"\$4 }' virbot/pos_contig_score.csv > pos_contig_score.tsv
+    """
+}
+
+process download_genomad_database {
+    label "process_single"
+    cpus 1
+    storeDir "${params.store_dir}/"
+    output:
+        path "genomad_db"
+    """
+    if [ -d "${params.store_dir}/genomad_db" ]
+    then
+        echo "Database already downloaded"
+    else
+        genomad download-database .
+    fi
+    """
+}
+process run_genomad {
+
+	label "process_medium"	
+
+	//UPLOAD ENV
+	conda "/localdisk/home/s2420489/conda/genomad.yml"
+	container "biowilko/scylla@${params.wf.container_sha}"
+
+	input:
+        val unique_id
+        path contigs, name: 'filtered_contigs.fa'
+        path genomad_db
+
+    output:
+        path "genomad/filtered_contigs_summary/filtered_contigs_virus.fna", emit: contigs
+	    path "tax_assignments.tsv", emit: taxonomy
+
+    script:
+    """
```

Comment:
What on earth is this indentation?????

Reply:
And this is one that doesn't appear as indented in my editor. Weird